[inductor] Update JIT Inductor cpp wrapper entry function signature

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #119280

Summary: Change JIT Inductor cpp wrapper entry function to use similar signature as AOTInductor, i.e. using an array of AtenTensorHandle instead of a vector of at::Tensor as the inputs and return output through a pointer. This makes it easier to consolidate the ABI compatible and non-compatible modes.

Differential Revision: [D53478825](https://our.internmc.facebook.com/intern/diff/D53478825)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

desertfire committed:
[inductor] Update JIT Inductor cpp wrapper entry function signature

Summary: Change JIT Inductor cpp wrapper entry function to use similar signature as AOTInductor, i.e. using an array of AtenTensorHandle instead of a vector of at::Tensor as the inputs and return output through a pointer. This makes it easier to consolidate the ABI compatible and non-compatible modes.

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Update JIT Inductor cpp wrapper entry function signature"


Summary: Change JIT Inductor cpp wrapper entry function to use similar signature as AOTInductor, i.e. using an array of AtenTensorHandle instead of a vector of at::Tensor as the inputs and return output through a pointer. This makes it easier to consolidate the ABI compatible and non-compatible modes.

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D53478825).

desertfire committed:
Update on "[inductor] Update JIT Inductor cpp wrapper entry function signature"


Summary: Change JIT Inductor cpp wrapper entry function to use similar signature as AOTInductor, i.e. using an array of AtenTensorHandle instead of a vector of at::Tensor as the inputs and return output through a pointer. This makes it easier to consolidate the ABI compatible and non-compatible modes.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

Differential Revision: [D53478825](https://our.internmc.facebook.com/intern/diff/D53478825)

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Update JIT Inductor cpp wrapper entry function signature"


Summary: Change JIT Inductor cpp wrapper entry function to use similar signature as AOTInductor, i.e. using an array of AtenTensorHandle instead of a vector of at::Tensor as the inputs and return output through a pointer. This makes it easier to consolidate the ABI compatible and non-compatible modes.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

Differential Revision: [D53478825](https://our.internmc.facebook.com/intern/diff/D53478825)

[ghstack-poisoned]

peterbell10 commented on torch/_inductor/codegen/wrapper.py:
```diff
@@ -2072,137 +2082,136 @@ def get_output_refs(self):
+                    output_expr = output
+                self.wrapper_call.writeline(
+                    f"output_handles[{idx}] = reinterpret_cast<AtenTensorHandle>("
+                    + f"new at::Tensor({output_expr}));"
```
Not this PR's fault, but why is `AtenTensorHandle` created from `new at::Tensor(x)` instead of `at::Tensor(x).unsafeReleaseTensorImpl()`? This would avoid an unnecessary heap allocation.

peterbell10 replied:
Or for returns specifically we can avoid the refcount bump as well.

desertfire replied:
Good suggestion. The main reason was I mapped AtenTensorHandle to at::Tensor pointer for simplicity, but it's an improvement we were thinking to do, https://github.com/pytorch/pytorch/blob/f3645fc38bedaa1b357d99d802728c99ee065667/torch/csrc/inductor/aoti_torch/c/shim.h#L53-L58 . I will come up with separate PRs for that.

desertfire committed:
Update on "[inductor] Update JIT Inductor cpp wrapper entry function signature"


Summary: Change JIT Inductor cpp wrapper entry function to use similar signature as AOTInductor, i.e. using an array of AtenTensorHandle instead of a vector of at::Tensor as the inputs and return output through a pointer. This makes it easier to consolidate the ABI compatible and non-compatible modes.

Differential Revision: [D53478825](https://our.internmc.facebook.com/intern/diff/D53478825)

[ghstack-poisoned]

desertfire commented on torch/_inductor/codegen/wrapper.py:
```diff
@@ -2072,137 +2082,136 @@ def get_output_refs(self):
         ]
 
     def generate_return(self, output_refs):
-        if V.graph.aot_mode:
```
This part is hard to review with eyeball. It's because of removing this checking caused an indentation difference.

chenyang78 reviewed (approved):
lgtm. Thanks!

desertfire committed:
Update on "[inductor] Update JIT Inductor cpp wrapper entry function signature"


Summary: Change JIT Inductor cpp wrapper entry function to use similar signature as AOTInductor, i.e. using an array of AtenTensorHandle instead of a vector of at::Tensor as the inputs and return output through a pointer. This makes it easier to consolidate the ABI compatible and non-compatible modes.

Differential Revision: [D53478825](https://our.internmc.facebook.com/intern/diff/D53478825)

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D53478825).
